runtime/gc_base/GCExtensions.cpp

```diff
@@ -249,18 +249,17 @@ MM_GCExtensions::computeDefaultMaxHeapForJava(bool enableOriginalJDK8HeapSizeCom
 {
 	OMRPORT_ACCESS_FROM_OMRVM(_omrVM);
 	uintptr_t maxMemoryValue = memoryMax;
-	if (OMR_CGROUP_SUBSYSTEM_MEMORY == omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_MEMORY)) {
-		if (omrsysinfo_cgroup_is_memlimit_set()) {
-			/* If running in a cgroup with memory limit > 1G, reserve at-least 512M for JVM's internal requirements
-			 * like JIT compilation etc, and extend default max heap memory to at-most 75% of cgroup limit.
-			 * The value reserved for JVM's internal requirements excludes heap. This value is a conservative
-			 * estimate of the JVM's internal requirements, given that one compilation thread can use up to 256M.
-			 */
+	if ((OMR_CGROUP_SUBSYSTEM_MEMORY == omrsysinfo_cgroup_are_subsystems_enabled(OMR_CGROUP_SUBSYSTEM_MEMORY) && omrsysinfo_cgroup_is_memlimit_set())
+		|| testContainerMemLimit) {
+		/* If running in a cgroup with memory limit > 1G, reserve at-least 512M for JVM's internal requirements
+		 * like JIT compilation etc, and extend default max heap memory to at-most 75% of cgroup limit.
+		 * The value reserved for JVM's internal requirements excludes heap. This value is a conservative
+		 * estimate of the JVM's internal requirements, given that one compilation thread can use up to 256M.
+		 */
```

Comment:
I think we should extend comment and explain what `testContainerMemLimit` is used for.

Reply:
However, mentioning softmx is a bit too specific. I prefer more generic comment like "Also used in test farms to force container limits, where it's non-trivial to actually create/config containers for each test.

Reply:
I can follow that